test/core/origin_core/commit_metablock.js

```diff
@@ -0,0 +1,527 @@
+             expectedRequiredWeight
+        );
+
+        let expectedHeight = 1;
+
+        OriginCoreUtils.assertCommitMetaBlock(
+             events,
+             expectedHeight,
+             kernelHash,
+             vote.transitionHash,
+             expectedRequiredWeight,
+             expectedVerifiedWeight
+        );
+
+        let expectedHead = MetaBlockUtils.hashMetaBlock(kernelHash,vote.transitionHash);
+        let head = await originCore.head.call();
+
+
+        assert.equal(
+          expectedHead,
+          head,
+          `Expected meta-block head ${expectedHead} is not equal to actual head ${head}`
+        );
+    });
+
+    it('should not commit already committed meta-block', async function () {
+        let stakeAddress = await originCore.stake.call();
+
+        initialDepositors = [
+            accounts[2],
+            accounts[3],
+            accounts[4],
+        ];
+        initialValidators = [
+            accounts[5],
+            accounts[6],
+            accounts[7],
+        ];
+        initialStakes = [
+            new BN('100'),
+            new BN('100'),
+            new BN('100'),
+        ];
+
+        await OriginCoreUtils.initializeStakeContract(
+             stakeAddress,
+             ost,
+             tokenDeployer,
+             initialDepositors,
+             initialStakes,
+             initialValidators
+        );
+
+        let expectedVerifiedWeight = new BN(initialStakes[0]);
+        let expectedRequiredWeight = new BN(200);
+
+        await OriginCoreUtils.verifyVote(
+             initialStakes[0],
+             initialValidators[0],
+             vote,
+             originCore,
+             kernelHash,
+             expectedVerifiedWeight,
+             expectedRequiredWeight
+        );
+
+        expectedVerifiedWeight = expectedVerifiedWeight.add(new BN(initialStakes[1]));
+
+        let events = await OriginCoreUtils.verifyVote(
+             initialStakes[1],
+             initialValidators[1],
+             vote,
+             originCore,
+             kernelHash,
+             expectedVerifiedWeight,
+             expectedRequiredWeight
+        );
+
+        let expectedHeight = 1;
+
+        OriginCoreUtils.assertCommitMetaBlock(
+             events,
+             expectedHeight,
+             kernelHash,
+             vote.transitionHash,
+             expectedRequiredWeight,
+             expectedVerifiedWeight
+        );
+
+        expectedVerifiedWeight = expectedVerifiedWeight.add(new BN(initialStakes[2]));
+
+         events = await OriginCoreUtils.verifyVote(
+             initialStakes[2],
+             initialValidators[2],
+             vote,
+             originCore,
+             kernelHash,
+             expectedVerifiedWeight,
+             expectedRequiredWeight
+        );
+
+         assert(
+              events.MetaBlockCommitted === undefined,
+              `Meta-block should only be committed once.`
+         );
+
+    });
+
+});
```

Comment:
It seems like a lot of code is duplicated across tests, not sure though.

Reply:
`      initialDepositors = [
            accounts[2],
            accounts[3],
            accounts[4],
        ];
        initialValidators = [
            accounts[5],
            accounts[6],
            accounts[7],
        ];
        initialStakes = [
            new BN('100'),
            new BN('100'),
            new BN('100'),
        ];
`

This part of the code is intentionally duplicated. This makes test cases more readable.  
Do you have a strong opinion on this? we can discuss this.

Reply:
All good